solr/core/src/java/org/apache/solr/search/grouping/distributed/responseprocessor/StoredFieldsShardResponseProcessor.java

```diff
@@ -38,6 +38,10 @@ public void process(ResponseBuilder rb, ShardRequest shardRequest) {
     SolrDocumentList docs = (SolrDocumentList)srsp.getSolrResponse().getResponse().get("response");
     String uniqueIdFieldName = rb.req.getSchema().getUniqueKeyField().getName();
 
+    if (rb.rsp.getReturnFields().getFieldRenames().get(uniqueIdFieldName) != null) {
+      // if id was renamed we need to use the new name
+      uniqueIdFieldName = rb.rsp.getReturnFields().getFieldRenames().get(uniqueIdFieldName);
+    }
```

Comment:
observation: for non-grouped queries the query component already has similar logic e.g. https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.8.1/solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java#L1187-L1190